.travis.yml

```
@@ -0,0 +1,23 @@
+dist: bionic
+
+sudo: required
+
+language: python
+
+python:
+  - 3.6
+  - 3.7
+  - 3.8
+
+before_script:
+  - pip install .
+
+script:
+  - python -c "import paperscraper"
+
+deploy:
+  provider: script
+  skip_cleanup: true
+  script: bash .travis/deploy.sh
+  on:
+    branch: master
```

```suggestion
  on:
    tags: true
    branch: master
```
This gives a bit more control over deployment to pypi, as tags need to be pushed explicitly.
Not sure that is in the same spirit as the decision to protect master from direct commits though.

>I implemented the automated upload on PyPI for every commit on master that is updating __version__ in paperscraper/__init__.py (settings here).

@drugilsberg I don't get how is this implemented. Is it just failing the deploy because of an existing version? That would mean only release commits (changing version) will succeed on master.

---

That is exactly it, this is what I found in travis and adapted twine to not fail if the version exist.

---

No strong feelings change as needed.

---

Ah I see, the skip-existing-dist would probably also prevent the different python jobs from failing.